### PR TITLE
Improvements for database setup docs

### PR DESCRIPTION
**Pull request by mik-laj**
- First of all, the previous version had a title that did not describe its content well.  I corrected it which will make it easier to find this page. 
- I have added sections that describe the requirements for the supported database versions. Previously, there was only a link to HA requirements, but no description for the basic configuration.
- I have moved the sections on database URI to the top of this document as this knowledge is needed in the following sections.
- I have created separate sections for MySQL and PostgreSQL because users who use MySQL are not interested in the information for PostgresSQL and vice versa. This also allowed us to delete various blocks of notes that were difficult to format and read.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: docs/apache-airflow/howto/set-up-database.rst [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+the example below.
+
+.. code-block:: bash
+
+    $ airflow config get-value core sql_alchemy_conn
+    sqlite:////tmp/airflow/airflow.db
+
+The exact format description is described in the SQLAlchemy documentation, see `Database Urls <https://docs.sqlalchemy.org/en/14/core/engines.html>`__. We will also show you some examples below.
+
+Set up a MySQL
```
Comment (kaxil) [inline docs/apache-airflow/howto/set-up-database.rst]:
```suggestion
Set up MySQL
```

or

```suggestion
Setting up a MySQL Database
```

File: docs/apache-airflow/howto/set-up-database.rst [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+.. code-block:: text
+
+   mysql+mysqlconnector://<user>:<password>@<host>[:<port>]/<dbname>
+
+However if you want to use other drivers visit the `MySQL Dialect <https://docs.sqlalchemy.org/en/13/dialects/mysql.html>`__  in SQLAlchemy documentation for more information regarding download
+and setup of the SqlAlchemy connection.
+
+Set up a PostgreSQL
+-------------------
```
Comment (kaxil) [inline docs/apache-airflow/howto/set-up-database.rst]:
```suggestion
Set up a PostgreSQL Database
-----------------------------
```

or 

```suggestion
Set up PostgreSQL
-----------------------------
```

File: docs/apache-airflow/howto/set-up-database.rst [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+You need to create a database and a database user that Airflow will use to access this database.
+In the example below, a database ``airflow_db`` and user  with username ``airflow_user`` with password ``airflow_pass`` will be created
+
+.. code-block:: sql
+
+   CREATE DATABASE airflow_db;
+   CREATE USER airflow_user WITH PASSWORD 'airflow_user';
+   GRANT ALL PRIVILEGES ON DATABASE airflow_db TO airflow_user;
+
+You need to create a database and a user that Airflow will use to access this database.
```
Comment (kaxil) [inline docs/apache-airflow/howto/set-up-database.rst]:
```suggestion
```

already mentioned before the code-block

**Review by kaxil — COMMENTED** [review]
Minor changes suggested or else LGTM

**Review by potiuk — APPROVED** [review]

**Comment by mik-laj** [comment]
@kaxil Updated. See:  https://github.com/apache/airflow/pull/13696/commits/1662aa05940b388a837d96065a15a595f2125768

**Review by kaxil — APPROVED** [review]

**Pull request merged**